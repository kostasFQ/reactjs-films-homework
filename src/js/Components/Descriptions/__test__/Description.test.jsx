import React from 'react';
import renderer from 'react-test-renderer';
import Description from '../index';

const fakeFilm = {
  pict: 'sss',
  title: 'The jungle book',
  genres: ['Adventure', 'Drama', 'Family', 'Fantasy'],
  duration: '1h 46m',
  rating: 4.8,
  preview: 'There ',
};

test('test component', () => {
  const render = renderer.create(
    <Description {...fakeFilm} />,
  );

  const component = render.toJSON();

  expect(component).toMatchSnapshot();
  expect(component.type).toBe('div');
});
