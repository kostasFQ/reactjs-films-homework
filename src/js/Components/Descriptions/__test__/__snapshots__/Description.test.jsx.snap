// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DESCRIPTION test DESCRIPTION component 1`] = `
<div
  className="container"
>
  <div
    className="container"
  >
    <h1
      className="h1"
    >
      VENOM
    </h1>
    <ul
      className="list"
    >
      <li
        className="item"
      >
        Action
      </li>
      <li
        className="item"
      >
        Crime
      </li>
      <li
        className="item"
      >
        Horror
      </li>
      <li
        className="item"
      >
        Science Fiction
      </li>
    </ul>
    <div
      className="container"
    >
      <div
        className="ratingBox"
      >
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
      </div>
      <div
        className="x"
        style={
          Object {
            "width": "132px",
          }
        }
      >
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
      </div>
      <div
        className="container"
      >
        6.6
      </div>
    </div>
  </div>
  <div
    className="buttonsBlock"
  >
    <button
      className="container"
      onClick={[Function]}
      type="button"
    >
      Watch Now
    </button>
    <button
      className="container"
      onClick={[Function]}
      type="button"
    >
      View Info
    </button>
  </div>
</div>
`;

exports[`DESCRIPTION test DESCRIPTION component click "View Info" 1`] = `
<div
  className="container"
>
  <div
    className="container"
  >
    <h1
      className="h1"
    >
      VENOM
    </h1>
    <ul
      className="list"
    >
      <li
        className="item"
      >
        Action
      </li>
      <li
        className="item"
      >
        Crime
      </li>
      <li
        className="item"
      >
        Horror
      </li>
      <li
        className="item"
      >
        Science Fiction
      </li>
    </ul>
    <div
      className="container"
    >
      <div
        className="ratingBox"
      >
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
      </div>
      <div
        className="x"
        style={
          Object {
            "width": "132px",
          }
        }
      >
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
        <img
          alt="star"
          className="star"
          src="test-file-stub"
        />
      </div>
      <div
        className="container"
      >
        6.6
      </div>
    </div>
  </div>
  <div
    className="buttonsBlock"
  >
    <button
      className="container"
      onClick={[Function]}
      type="button"
    >
      Watch Now
    </button>
    <button
      className="container"
      onClick={[Function]}
      type="button"
    >
      View Info
    </button>
  </div>
  <div
    className="prevField"
  >
    <div
      className="container"
    >
      <div
        className="buttonsBlock"
      >
        <div>
          When Eddie Brock acquires the powers of a symbiote, he will have to release his alter-ego "Venom" to save his life.
        </div>
      </div>
    </div>
  </div>
</div>
`;
